include/deal.II/base/parameter_handler.h

```diff
@@ -1043,7 +1056,29 @@ class ParameterHandler : public Subscriptor
    * input.
    */
   virtual void
-  parse_input_from_json(std::istream &input, const bool skip_undefined = false);
+  parse_input_from_json(std::istream &input,
+                        const bool    skip_undefined                  = false,
+                        const bool assert_mandatory_entries_are_found = false);
+
+  /**
+   * Parse input from a specified @param parameter_file independently of the type
+   * of input file (prm, xml, json) being used. The code path selected by this
+   * function is extracted from the ending of the filename, so the user has to
+   * make sure that the content of the input file is consistent with its name.
+   * The user can specify whether parameters in the input file not added to the
+   * parameter handler will be skipped (enables partial parsing), and whether
+   * the code will assert that all parameters of the parameter handler declared
+   * with flag `has_to_be_set=true` are indeed found in the input file. If the
+   * function is called with `skip_undefined=true`, it is recommended to also
+   * set `assert_mandatory_entries_are_found=true`. For example, this ensures
+   * that parameters with typos in the input file will not be skipped, while
+   * such mistakes would otherwise remain unrecognized.
+   */
+  void
+  parse_input_any_type(const std::string parameter_file,
```

Comment:
`from_file` also holds for other `parse_input_...()` functions, so I don't think it is a better description for this particular function. I originally used just `parse_input()` but due to the i) function overloading of many parse_input-functions and ii) using default values for some function parameters, the code did no longer compile because of ambiguities (what makes things more complicated, there are dervied classes MultipleParameterLoop, ParameterAcceptor). I would agree on ``...from_any_file()``?

Reply:
Indeed there is a `parse_input()`-function which takes a file name (https://github.com/dealii/dealii/blob/4d7085ea23b89032b6af6e53ce767f25ae207953/source/base/parameter_handler.cc#L509-L519). Is there a reason why you did not extend that function, but created a new function. Am I missing something?

Reply:
In a notation consistent with the other parse_input() functions, one would probably call this function ``pasrse_input_from_prm()``? To explain the names, I can only speculate that this has been the first function and json-, xml-versions were added later.

I intentionally created a new one because I would otherwise change the behavior of this function, do you agree? For example, I think this parse_input() function can be called without the .prm file ending. So codes calliing this function without file ending would then run into the `Assert()` implemented in the new function `parse_input_any_type()` if I had extended this existing function.

Reply:
Thanks for the explanation! In think a minor change in behavior would be acceptable. An option would be that files ending with `.json` or `.xml` are treated as such and all other files (including the files without ending) are treated as `prm` files. The probability that someone is using the suffix `.json`/`.xml` for a `prm`-file is quite low, do yo agree?

Reply:
I see your point. On the other hand, I actually would like to not have to make workarounds with implicit assumptions (like all other file endings are interpreted as prm; other file endings might be added in the future). We will right now understand why it is implemented like this, but this will soon get blurred and others will encounter a situation similar to the current one that  needs to be improved. I would suggest that we open an issue with the question whether we are willing to make incompatible changes to `ParameterHandler` for a better design/interface. For now, to have all options for the future and If we don't like the ending `...any_type()` because this has of course historical reasons (I fully agree in this point), I would suggest that we for now agree on another generic name like `parse_parameters()`, `parse_file()`,  `parse_input_file()` (I think there are enough options) and deprecate/remove other parse functions in the long run depending on the outcome of the issue. For me it is not worth combining the new function with the existing `parse_input()` function just for reserving the name. Why? Then we still have to get rid of the other four `parse_input()` functions. This will otherwise lead to misunderstandings: For example, people will then ask themselves why is there a function `parse_input_from_json()` if `parse_input()` can handle everything. So I think we actually have to get rid of the multitude of `parse_input_...()` functions. If one can convince to deprecate/remove `parse_input()` functions with all the associated incompatibilities, from that perspective I think it does not make a big difference whether these are four or five functions.

Reply:
I think the correct way forward would be to change `parse_input. I had to read through all comments to understand what was going on.... `parse_input_any_type` does not ring any bell. I thought the function would allow to parse any parameter type, not any file type. It is clearly wrong that the `parse_input` function that takes a string would happily read a `file.xml` in the same way as `file.prm`. 
In fact, now it would throw, and a user would have to understand that `parse_input_any_type` and `parse_input` are actually the exact same thing for `prm` and differ only for other file types, where one does not work, and the other has just a more complicated name. 
My two cents would be to just change the behavior of the currently existing function (by **extending** it). It would be backward compatible, and would be much easier to understand than having `parse_input_any_type` that takes a string. I argue that `parse_input_from_XXX` still makes sense for input streams, since that’s the only possibility we have, but we should not make easy things complicated when it is unnecessary. I see no “implicit” assumptions, provided that the methods are well documented.

Reply:
The implicit assumptions and incompatibilities arise when you take the comments of @kronbichler and @peterrum together. I take this as an authorization and will change it accordingly.